cpp/daal/src/data_management/train_test_split.cpp

```diff
@@ -39,6 +41,123 @@ typedef daal::data_management::NumericTable::StorageLayout NTLayout;
 const size_t BLOCK_CONST      = 2048;
 const size_t THREADING_BORDER = 8388608;
 
+const size_t MT19937_NUMBERS        = 624;
+const size_t MT19937_SIZE           = 631;
+const size_t MT19937_NUMBERS_OFFSET = 5;
+
+size_t genSwapIdx(size_t i, unsigned int * randomNumbers, size_t & rnIdx)
+{
+    uint32_t bitMask = i;
+    bitMask |= bitMask >> 1;
+    bitMask |= bitMask >> 2;
+    bitMask |= bitMask >> 4;
+    bitMask |= bitMask >> 8;
+    bitMask |= bitMask >> 16;
+
+    size_t j = randomNumbers[rnIdx] & bitMask;
```

Comment:
minnor: Use const when you do not change data/variables

Reply:
But it actually changes